### PR TITLE
Add organization flag and deprecate owner instead

### PR DESCRIPTION
**Pull request by pipo02mix**
There is no official issue but SIG UX discussion and make a lot of sense to align the naming here https://gigantic.slack.com/archives/C13MG9F0W/p1631274549023400

### CODE REVIEW

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -129,8 +133,6 @@ func (f *flag) Validate() error {
 				return microerror.Maskf(invalidFlagError, "--%s must be a valid subnet size (20, 21, 22, 23, 24 or 25)", flagControlPlaneSubnet)
 			}
 		}
-
-		return nil
```
Comment (pipo02mix) [inline cmd/template/cluster/flag.go:133]: IMO this was a bug preventing rest of code to be executed (without reason)

**Review by MarcelMue — APPROVED** [review]
Makes sense to me.

Just for coordination: If either this or #463 merged first, we need to resolve some merge conflicts :)

**Review by marians — CHANGES_REQUESTED** [review]
Currently the changelog entry mentions `--organization` as the new flag, while in code it's `--org`.

I would vote for `--organization` in code, as "organization" is the proper terminology we use and I recommend against introducing another medium-short form in flag names.

File: cmd/template/cluster/flag.go [hunk 2]
```diff
@@ -26,7 +26,8 @@ const (
 	flagMasterAZ            = "master-az" // TODO: Remove some time after August 2021
 	flagName                = "name"
 	flagOutput              = "output"
-	flagOwner               = "owner"
+	flagOrganization        = "org"
```
Comment (pipo02mix) [inline cmd/template/cluster/flag.go:29]:
```suggestion
	flagOrganization        = "organization"
```

**Comment by marians** [comment]
@pipo02mix Did you forget to push? I see no changes.

**Comment by pipo02mix** [comment]
@marians hehe you were right 🙈

File: cmd/template/nodepool/flag.go [hunk 3]
```diff
@@ -117,6 +120,7 @@ func (f *flag) Init(cmd *cobra.Command) {
 	// To be removed around December 2021
 	_ = cmd.Flags().MarkDeprecated(flagClusterIDDeprecated, "use --cluster-name instead")
 	_ = cmd.Flags().MarkDeprecated(flagNodepoolNameDeprecated, "use --description instead")
+	_ = cmd.Flags().MarkDeprecated(flagOwner, "please use --org instead.")
```
Comment (marians) [inline cmd/template/nodepool/flag.go:123]:
```suggestion
	_ = cmd.Flags().MarkDeprecated(flagOwner, "please use --organization instead.")
```

File: cmd/template/cluster/flag.go [hunk 4]
```diff
@@ -139,8 +141,18 @@ func (f *flag) Validate() error {
 		}
 	}
 
-	if f.Owner == "" {
-		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagOwner)
+	// TODO: Remove the flag completely some time after December 2021
+	if f.Owner == "" && f.Organization == "" {
+		return microerror.Maskf(invalidFlagError, "--%s or --%s must not be empty", flagOwner, flagOrganization)
```
Comment (marians) [inline cmd/template/cluster/flag.go:146]: Let's use the desired flag name only here

```suggestion
		return microerror.Maskf(invalidFlagError, "--%s must not be empty", flagOrganization)
```

File: cmd/template/cluster/flag.go (hunk 1 above)
Comment (marians) [reply cmd/template/cluster/flag.go:133]: 🤷

**Review by marians — APPROVED** [review]

**Comment by marians** [comment]
Thanks for hanging in there!

**Pull request merged**